src/predict_request_validation_utils.cpp

```diff
@@ -131,6 +153,13 @@ Status RequestValidator<KFSRequestType, KFSInputTensorType, KFSInputTensorIterat
         }
     }
     if (request.inputs_size() > 0 && expectedNumberOfInputs == static_cast<size_t>(request.inputs_size())) {
+        if (!request.raw_input_contents().empty()) {
```

Comment:
Why do we need this check? From my understanding this call is supposed to check if there is correct number of inputs, but this does something more.